source/common/config/grpc_mux_impl.cc

```diff
@@ -167,9 +167,9 @@ void GrpcMuxImpl::onReceiveMessage(std::unique_ptr<envoy::api::v2::DiscoveryResp
   const std::string& type_url = message->type_url();
   ENVOY_LOG(debug, "Received gRPC message for {} at version {}", type_url, message->version_info());
   if (api_state_.count(type_url) == 0) {
-    ENVOY_LOG(warn, "Ignoring unknown type URL {}", type_url);
-    // TODO(yuval-k): This should never happen. consider dropping the stream as this is a protocol
-    // violation
+    ENVOY_LOG(warn, "Protocol Error: gRPC Message for type URL {} received before subscription",
```

Comment:
@ramaraochavali can you elaborate a bit on the motivation here? I agree that this is a reasonable behavior, trying to understand the context for the change. Thanks.

Reply:
@htuch given the discussion we are having in #4716 , since this does not seem to be a long term solution, should I close this PR? or do we want to move forward because it is a reasonable behavior for now? I am ok either way. LMK.

Reply:
Yeah, should probably follow this up in #4716 and close for now. While I think what you have here is reasonable behavior, I think this is a pretty risky change; some control planes might be relying on the silent failure and this could break them. It doesn't hurt to have Envoy be more permissive.

Reply:
Ok. Sure.. Will close it.